homeassistant/components/snmp/util.py

```diff
@@ -0,0 +1,65 @@
+"""Support for displaying collected data over SNMP."""
+
+from __future__ import annotations
+
+from functools import cache
+
+from pysnmp.hlapi.asyncio import (
+    CommunityData,
+    ContextData,
+    ObjectIdentity,
+    ObjectType,
+    SnmpEngine,
+    Udp6TransportTarget,
+    UdpTransportTarget,
+    UsmUserData,
+)
+from pysnmp.hlapi.asyncio.cmdgen import vbProcessor
+from pysnmp.smi.builder import MibBuilder
+
+from homeassistant.core import HomeAssistant
+
+type RequestArgsType = tuple[
+    SnmpEngine,
+    UsmUserData | CommunityData,
+    UdpTransportTarget | Udp6TransportTarget,
+    ContextData,
+    ObjectType,
+]
+
+
+@cache
+def snmp_engine() -> SnmpEngine:
```

Comment:
What do you think about using a common SnmpEngine for Brother and SNMP integration in the future? For the needs of Brother SnmpEngine it is a sigleton.
https://github.com/home-assistant/core/blob/cdcf091c9c728371c9ca999b94dcc9dd149652f5/homeassistant/components/brother/utils.py#L19-L33